test/distributed/test_inductor_collectives.py

```diff
@@ -712,6 +712,30 @@ def func(inp, out, *, pg):
         assert counter.op_count == 3
         assert same(outputs, correct_outputs)
 
+    def test_dynamo_rewrite_dist_allreduce(self):
+
+        def func(tensor, pg):
```

Comment:
I'm a little confused about the expected functionality for what params are allowed to be compiled. Trying to pass `op="sum"` or `op=dist..ReduceOp.Sum` generates an exception in this and other re-mapped methods. 

Is this expected? 

@wconstab

Reply:
Not supporting `op` is probably just an oversight/lazyness.  We may only be able to support the string variant today bc of serialization/IR but i'm not sure about that and maybe the enum also works.

But to make it work, you'll have to update the dynamo code around here to parse that extra arg

```
    def call_function(
        self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
    ) -> "VariableTracker":
        # call_function must check any unsupported arguments and graph-break.
        # It's safe to assume args/kwargs from orig_fn map 1:1 to args/kwargs of remapped_fn,
        # since that's the contract for putting a mapping in `traceable_collective_remaps`
        if kwargs.get("async_op", False):
            unimplemented(
                f"CollectiveFunctionRewriteVariable can't support async_op=True for {self.fn}"
            )
        return self.replacement_var.call_function(tx, args, kwargs)
        ```